server/utils/bassAddressState.js

```diff
@@ -0,0 +1,121 @@
+const { taskStates } = require('../services/config/taskStates')
+const { getIn, isEmpty, lastItem } = require('./functionalHelpers')
+
+module.exports = { getBassAreaState, getBassState, getBassRequestState }
+
+function getBassRequestState(licence) {
+  const bassRequestAnswer = getIn(licence, ['bassReferral', 'bassRequest', 'bassRequested'])
+  const addressProposedAnswer = getIn(licence, ['proposedAddress', 'addressProposed', 'decision'])
+
+  const bassReferralNeeded = bassRequestAnswer === 'Yes' && addressProposedAnswer === 'No'
+  const bassAreaSpecified = getIn(licence, ['bassReferral', 'bassRequest', 'specificArea']) !== 'No'
+  const bassRequest = getState()
+
```

Comment:
getTaskState might be a better name than getState...

Reply:
yes - I don't like that it's an inner function either, this is just a whole sale copy over